vendor/bat-native-ledger/src/bat_client.cc

```diff
@@ -484,7 +509,7 @@ void BatClient::setGrantCallback(bool success,
 
   if (!success) {
     if (statusCode == 403) {
-      ledger_->OnGrantFinish(ledger::Result::CAPTCHA_FAILED, grant);
+      ledger_->OnGrantFinish(isAndroid ? ledger::Result::SAFETYNET_ATTESTATION_FAILED : ledger::Result::CAPTCHA_FAILED, grant);
```

Comment:
mixing up the logic for the attestation between ledger and components/brave_rewards seems problematic to me. why would a return status code for this be an attestation failure for android?

Reply:
This is for UI handler, so it could understand how to process the error code. CAPTCHA_FAILED - user can repeat attempt. SAFETYNET_ATTESTATION_FAILED - no need to repeat anything.